### ISSUE TITLE
building for iOS-armv7 but attempting to link with file built for iOS-arm64   Undefined symbols for architecture armv7

### ISSUE THREAD

**Issue by KelvinForteta** [helpscout/beacon-ios-sdk#147]
Good day,

I try to integrate Beacon into our company app, everything works well on android, but on iOS after integrating and following all the steps in the documentation, when I ran it on Simulator it works fine apart from not display the close and back button, then I decided to build it for production and I got the following error:

`Error output from Xcode build:
↳
    ** BUILD FAILED **

Xcode's output:
↳
    ld: warning: ignoring file
    /Users/kelvinforteta/Documents/Projects/MoneyMie/moneymie_bank/build/ios/Release-iphoneos/XCFrameworkIntermediates/Beaco
    n/Beacon.framework/Beacon, building for iOS-armv7 but attempting to link with file built for iOS-arm64
    Undefined symbols for architecture armv7:
      "_OBJC_CLASS_$_HSBeacon", referenced from:
          objc-class-ref in SwiftFlutterHelpScoutPlugin.o
      "_OBJC_CLASS_$_HSBeaconUser", referenced from:
          objc-class-ref in SwiftFlutterHelpScoutPlugin.o
      "_OBJC_CLASS_$_HSBeaconSettings", referenced from:
          objc-class-ref in SwiftFlutterHelpScoutPlugin.o
    ld: symbol(s) not found for architecture armv7
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Using build description from disk

Encountered error while building for device.`

SDK version : 2.1.2
The framework is Flutter using method channel

I have searched for a solution but haven't gotten any yet.

**Comment by scottyab** [comment]
Hey @KelvinForteta, thanks for raising an issue. Are you attempted to integrate the Beacon SDK into a Flutter app?

**Comment by KelvinForteta** [comment]
Yes, it is a flutter app

**Comment by ScottPetit** [comment]
Hey 👋 @KelvinForteta,

So since Beacon is iOS 11+ we actually don't/can't build for armv7 for the SDK since iOS 11 is when Apple deprecated support for armv7.  Is it possible for you to update the valid architectures of your project to exclude armv7 and attempt to rebuild for device again.  Also I suppose it's possible you're specifying `VALID_ARCHS` and building on an older device like an iPhone 4s in which case I'm not sure what the next steps would be aside from trying to build to a more modern device.

**Comment by KelvinForteta** [comment]
Ok, I will do just that and give you feedback.

**Comment by scottyab** [comment]
Closing this issue as we (and Apple) don't support armv7. Please do create a new issue if there's anything else we can help with. Thanks

**Comment by KelvinForteta** [comment]
Thanks for your help, it worked.
